p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -315,9 +308,7 @@ private boolean shouldTransmitPayloadToPeer(Capabilities peerCapabilities, Netwo
                 doAdd = true;
             }
             if (doAdd) {
-                boolean notContained = lookupSet.add(protectedStoragePayload.hashCode());
```

Comment:
@chimp1984 I could use your prior knowledge for the behavior. Do you know of any cases where we intentionally want/expect the payload hash and the hashCode to be different?

Reply:
It deals with the case that there are multiple `ProtectedStorageEntry` objects with the same `ProtectedStoragePayload`. Not 100% sure if there are valid cases for that as the map in `P2PDataStorage` should not contain entries with the same `ProtectedStoragePayload`. There was the old `TradeStatistics` which was a `ProtectedStoragePayload` but that is not used anymore. The new is a `PersistableNetworkPayload`. For `TradeStatistics2` objects it is valid that we receive it 2 times (both traders are publishing it) in a different envelopes.

So I am not aware of any actual use case, but maybe it is better to keep that logic (maybe add a comment to make it easier to understand) to be able to support such a use case in future and to not break things if we have overlooked an existing use case. But I leave it to your consideration. You know that domain already better than myself... ;-)